src/test/java/uk/gov/hmcts/reform/translate/service/DictionaryServiceTest.java

```diff
@@ -218,12 +241,180 @@ void shouldUpdateADictionaryForUserWithoutManageTranslationsRole() {
+            verify(securityUtils, times(1)).hasRole(anyString());
+            verify(dictionaryMapper, times(3)).modelToEntityWithTranslationUploadEntity(any(), any());
+            verify(dictionaryRepository, times(3)).save(any());
+
+        }
+
+        @Test
+        void shouldPutDictionaryRoleCheckForAValidUserWithLoadTranslationRole() {
+            final Dictionary dictionaryRequest = getDictionaryRequestWithoutABody(1, 2);
+            final DictionaryEntity dictionaryEntity =
+                createDictionaryEntity("english_1", "translated_1");
+            given(applicationParams.getPutDictionaryS2sServicesBypassRoleAuthCheck())
+                .willReturn(Arrays.asList(DEFINITION_STORE));
+            given(securityUtils.hasAnyOfTheseRoles(anyList())).willReturn(true);
+            given(securityUtils.getServiceNameFromS2SToken(CLIENTS2S_TOKEN)).willReturn(XUI);
+
+            given(dictionaryRepository.findByEnglishPhrase(any())).willReturn(Optional.of(dictionaryEntity));
+            given(securityUtils.getUserInfo()).willReturn(getUserInfoWithManageTranslationsRole());
+            given(securityUtils.hasRole(anyString())).willReturn(true);
+            dictionaryService.putDictionaryRoleCheck(CLIENTS2S_TOKEN);
+            dictionaryService.putDictionary(dictionaryRequest);
+
+            verify(dictionaryRepository, times(1)).findByEnglishPhrase(any());
+            verify(securityUtils, times(1)).hasRole(anyString());
+            verify(dictionaryRepository, times(1)).save(any());
+        }
+
+        @Test
+        void shouldPutDictionaryRoleCheckForAValidDefinitionStore() {
+
+            final Dictionary dictionaryRequest = getDictionaryRequestWithoutABody(1, 2);
+            final DictionaryEntity dictionaryEntity =
+                createDictionaryEntity("english_1", "translated_1");
+            given(applicationParams.getPutDictionaryS2sServicesBypassRoleAuthCheck())
+                .willReturn(Arrays.asList(DEFINITION_STORE));
+            given(securityUtils.getServiceNameFromS2SToken(CLIENTS2S_TOKEN)).willReturn(DEFINITION_STORE);
+
+            given(dictionaryRepository.findByEnglishPhrase(any())).willReturn(Optional.of(dictionaryEntity));
+            given(securityUtils.getUserInfo()).willReturn(getUserInfoWithManageTranslationsRole());
+            given(securityUtils.hasRole(anyString())).willReturn(true);
+            dictionaryService.putDictionaryRoleCheck(CLIENTS2S_TOKEN);
+            dictionaryService.putDictionary(dictionaryRequest);
+
+            verify(dictionaryRepository, times(1)).findByEnglishPhrase(any());
+            verify(securityUtils, times(1)).hasRole(anyString());
+            verify(dictionaryRepository, times(1)).save(any());
+
+        }
+
+
+        @Test
+        void shouldFailPutDictionaryRoleCheckForAValidUserWithOutAnyRole() {
+            lenient().when(securityUtils.hasRole(any())).thenReturn(true);
```

Comment:
If you need to add these to get your tests working because of global `BeforeEach`:  feel free to wrap previous get unit tests in a nested so this issue does not effect your own.   

NB: WLTS-3 is already doing this in pr #15.

Reply:
Resolved in c655d3d72b8f1560acb96d062f9650d0b82728af